core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -610,7 +610,7 @@ class StandaloneDynamicAllocationSuite
    * we submit a request to kill them. This must be called before each kill request.
    */
   private def syncExecutors(sc: SparkContext): Unit = {
-    val driverExecutors = sc.getExecutorStorageStatus
```

Comment:
You could replace this with code based on `sc.statusStore`.

Reply:
I have tried to use "sc.statusStore.executorList(true)" instead of sc.env.blockManager.master.getStorageStatus but the test failed.

Reply:
Failed how? The list kept by the block manager and by the status store should be the same, so if they differ, there's a problem somewhere.

Reply:
As only registered executors can be killed this part synchronised the executors known by the master and the driver (the missing executors from the driver was registered with some mock data). The reason behind was some performance. 

I have run tried to change the code to wait for the executors and it get very slow (one test took even 50 seconds long).